### PR TITLE
Third try start year

### PR DESCRIPTION
**Pull request by frmfisc**
new try still the snap shot problem

### CODE REVIEW

**Comment by frmfisc** [comment]
finally done pls review carefully

File: R/read.R [hunk 1]
```diff
@@ -1,4 +1,4 @@
-st_read_agnostic <- function(dir, start_year, sectors, risk_type) {
+st_read_agnostic <- function(dir, start_year, sectors, risk_type, start_year_two) {
```
Comment (bergalli) [inline R/read.R:1]: should be removed with file renaming Scenarios_AnalysisInput_2022.csv

File: R/read.R [hunk 2]
```diff
@@ -11,7 +11,7 @@ st_read_agnostic <- function(dir, start_year, sectors, risk_type) {
   out <- list(
     capacity_factors_power = capacity_factors_power,
     df_price = read_price_data(price_data_file(dir)),
-    scenario_data = read_scenario_data(scenario_data_file(dir, start_year)),
+    scenario_data = read_scenario_data(scenario_data_file(dir, start_year_two)),
```
Comment (bergalli) [inline R/read.R:14]: start_year should stay

File: R/run_lrisk.R [hunk 3]
```diff
@@ -169,7 +169,7 @@ run_lrisk_iteration <- function(args_list) {
 }
 
 # Avoid R CMD check NOTE: "Undefined global functions or variables"
-globalVariables(c(names(formals(run_lrisk)), "iter_var", "end_year"))
+globalVariables(c(names(formals(run_lrisk)), "iter_var", "end_year", "start_year_two"))
```
Comment (bergalli) [inline R/run_lrisk.R:172]: start_year_two should be removed

File: R/run_lrisk.R [hunk 4]
```diff
@@ -193,7 +193,7 @@ read_and_process_and_calc_lrisk <- function(args_list) {
 
   cat("-- Reading input data from designated input path. \n")
 
-  data <- st_read_agnostic(input_path, start_year = start_year, sectors = sectors_and_technologies_list$sectors, risk_type = "lrisk")
+  data <- st_read_agnostic(input_path, start_year = start_year, sectors = sectors_and_technologies_list$sectors, risk_type = "lrisk", start_year_two = start_year_two)
```
Comment (bergalli) [inline R/run_lrisk.R:196]: start_year_two to remove

File: R/run_trisk.R [hunk 5]
```diff
@@ -162,7 +162,7 @@ run_stress_test_iteration <- function(args_list) {
 }
 
 # Avoid R CMD check NOTE: "Undefined global functions or variables"
-globalVariables(c(names(formals(run_trisk)), "iter_var", "end_year"))
+globalVariables(c(names(formals(run_trisk)), "iter_var", "end_year", "start_year_two"))
```
Comment (bergalli) [inline R/run_trisk.R:165]: start_year_two to remove

File: R/run_trisk.R [hunk 6]
```diff
@@ -186,7 +186,7 @@ read_and_process_and_calc <- function(args_list) {
 
   cat("-- Reading input data from designated input path. \n")
 
-  data <- st_read_agnostic(input_path, start_year = start_year, sectors = sectors_and_technologies_list$sectors, risk_type = "trisk")
+  data <- st_read_agnostic(input_path, start_year = start_year, sectors = sectors_and_technologies_list$sectors, risk_type = "trisk", start_year_two = start_year_two)
```
Comment (bergalli) [inline R/run_trisk.R:189]: start_year_two to remove

File: R/process_data.R [hunk 7]
```diff
@@ -532,6 +532,14 @@ process_production_data <- function(data, start_year, end_year, time_horizon,
...
       )
     )
+  data_processed <- data_processed %>%
+    dplyr::mutate(
+      plan_sec_prod = dplyr::if_else(
+        is.na(.data$plan_sec_prod),
+        0,
+        .data$plan_sec_prod
+      )
+    )
```
Comment (bergalli) [inline R/process_data.R:542]: this has been removed but the wrangle_and_check_production_data() function is still in effect. We should remove it as what it does is filling missing year value, and we expect that there are none. Could be replaced by an assertr::verify operation to check this assumption ?

**Review by bergalli — COMMENTED** [review]
Unless I'm missing something , we should update the scenario data to 2022 and change the file name accordingly ?
Also, I'll review the production data code to be sure that there can be no missing years per company/business unit. Then we can remove wrangle_and_check_production_data() here

File: R/file_path_config.R [hunk 8]
```diff
@@ -8,8 +8,9 @@ price_data_file <- function(dir) {
   return(out)
 }
 
-scenario_data_file <- function(dir, start_year) {
-  out <- file.path(dir, glue::glue("Scenarios_AnalysisInput_{start_year}.csv"))
+scenario_data_file <- function(dir, start_year_two) {
+  start_year_two <- 2021
+  out <- file.path(dir, glue::glue("Scenarios_AnalysisInput_{start_year_two}.csv"))
```
[the remark this answers is not in the document]
Comment (frmfisc) [reply R/file_path_config.R:13]: no wrong the other way around...

File: R/read.R (hunk 1 above)
Comment (frmfisc) [reply R/read.R:1]: wrong...

File: R/file_path_config.R (hunk 8 above)
[the remark this answers is not in the document]
Comment (frmfisc) [reply R/file_path_config.R:13]: confusion cleared ill roll it back its release_year vs start_year

**Comment by bergalli** [comment]
the variable time_horizon_lookup (or time_horizon in geeral) should be removed everywhere it is referred and the code using it must be refactored to not use it anymore.
It seems it refers to the time horizon in the production data
It is confusing to have this variable as well as end_year and start_year

**Review by bergalli — APPROVED** [review]

**Pull request merged**